packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -1561,8 +1561,13 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
     if (!_shouldCreateInputConnection) {
       _closeInputConnectionIfNeeded();
     } else {
-      if (oldWidget.readOnly && _hasFocus)
+      if (oldWidget.readOnly && _hasFocus) {
```

Comment:
question: what is `oldwidget`?

Reply:
it is the original widget before the update

Reply:
This is a stateful widget, which means the `state` instance remains alive even if you build `EditableText` multiple times. So when you do:
```dart
Widget build(BuildContext context) {
  return EditableText(
    readOnly: true,
    ...
  );
}
```
^ this `EditableText` instance becomes the new value for `widget`, and the `EditableText` instance from the previous build is available in `oldWidget`.